python/hail/methods/impex.py

```diff
@@ -786,14 +786,16 @@ def grep(regex, path, max_count=100):
            min_partitions=nullable(int),
            reference_genome=nullable(reference_genome_type),
            contig_recoding=nullable(dictof(str, str)),
-           skip_invalid_loci=bool)
+           skip_invalid_loci=bool,
+           row_fields=sequenceof(enumeration('varid', 'rsid')))
```

Comment:
can you make the entry fields type checker an enumeration of 'GP', 'GT', and 'dosage'?

Reply:
already done in a different PR that's been merged: https://github.com/hail-is/hail/pull/3778